src/main/java/com/wojciechbarwinski/demo/legendary_warehouse/exceptions/ErrorResponse.java

```diff
@@ -0,0 +1,12 @@
+package com.wojciechbarwinski.demo.legendary_warehouse.exceptions;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+public class ErrorResponse<T>{
+    private int errorCode;
```

Comment:
Nie ma sensu zwracać tutaj takiego kodu - już mamy w samej odpowiedzi ;) 
A tak generujesz potencjalny rozjazd, bo co jak się developer walnie i będzie zwrócone HTTP 200, a tu będzie 422? :P
AAAle - możesz zrobić jak najbardziej customowe kody. Zwracając sam message dajesz małe pole do popisu klientom jak wyświetlić error ich clientom (np. na UI) - takie customowe kody są bardziej elastyczne, bo pozwalają podpiąć np. komunikat w różnych językach jak potrzeba.

W naszym wypadku:
 - albo wywalmy ten kod, message nam wystarczy
 - albo zamień to pole na Stringa i wrzuć tam jakiś customowy kod z nowego enuma
 -

Reply:
oki, wiec usunałem kod HTTP, ale do exceptiona dodałem pole message, wiec w miejscu rzucania błędu będzie podana wiadomość którą poźniej przekaże się do ErrorResponse. Czy tak może być? 
Zakładam ze prawie kazdy exception jakiąś wiadomośc powinien miec a w tym momencie ErrorResponse ją otrzymuje i z dowolnymi danymi przesyła dalej.